### PR TITLE
Add support for pagination to lists of files

### PR DESCRIPTION
**Pull request by stevemessick**
This does not include the generated code, which should make reviewing easier. Once this is merged and the final handler PR is in, I'll do a release and commit the new generated code.

### CODE REVIEW

File: src/kaggle/api/kaggle_api_extended.py
Comment (jmasukawa) [inline src/kaggle/api/kaggle_api_extended.py:3525]: nit: might be nicer to use array destructure assignment syntax?

```py
[owner_slug, model_slug, framework, instance_slug, version_number] = urls
```

File: src/kaggle/api/kaggle_api_extended.py
Comment (jmasukawa) [inline src/kaggle/api/kaggle_api_extended.py:4365]: are the types here in both conditional paths the same?

it looks like we'll end up with a list if files is present, otherwise we set a dict here. in practice idk if it matters, but we probably want this to be same type?

**Review by jmasukawa — COMMENTED** [review]
Nice! Overall this LG, but i'll let Jim have the final approval here.

File: src/kaggle/api/kaggle_api_extended.py
Comment (stevemessick) [reply src/kaggle/api/kaggle_api_extended.py]: I changed all the ones I added to be more Pythonic. (Didn't touch the others.)

File: src/kaggle/api/kaggle_api_extended.py
Comment (stevemessick) [reply src/kaggle/api/kaggle_api_extended.py:95]: Yes, I'm running it in the container. I pretty much have to if I want to work on the C# and Python together. I tried opening the project in IntelliJ (because the Python plugin for Rider is currently not operational), but quickly realized I'd have to go through and install all the dependencies again. I didn't want to do that, since I don't know how many are pre-installed in the container. I mention that because I thought that's where the reformatting occurred (but `yapf` is more likely). I did restore almost all the format changes.

**Review by jplotts — COMMENTED** [review]
Looking good!  Can you please revert the license removals?  Also, I'm a bit unsure about the models API w.r.t. framework/instance/versions being required or not (specific questions below).  Thanks for tackling this!

File: src/kaggle/api/kaggle_api_extended.py
Comment (stevemessick) [reply src/kaggle/api/kaggle_api_extended.py:95]: In the `Development` section, I'm adding this text. Let me know if there's any problem with it:

### Kaggle Internal

Obviously, this depends on Kaggle services. When you're extending the API and modifying
or adding to those services, you should be working in your Kaggle mid-tier development
environment. You'll run Kaggle locally, in the container, and test the Python code by
running it in the container so it can connect to your local testing environment.

Comment (stevemessick) [reply src/kaggle/api/kaggle_api_extended.py:95]: That's the `Development` section of `README.md`.

Comment (rosbo) [reply src/kaggle/api/kaggle_api_extended.py:95]: In case you are not aware, the releases for `kagglehub` and `kaggle-api` are done using Google Cloud Build. Anyone can make a release in a consistent environment. You just need to trigger Cloud Build with a CLI command.

Docmentation for `kagglehub`: https://g3doc.corp.google.com/cloud/kaggle/models/g3doc/kagglehub.md#step-2-create-a-new-release

For `kaggle-api`,  it is documented here: https://g3doc.corp.google.com/company/teams/kaggle/cli/index.md#release

File: src/KaggleSwagger.yaml
Comment (stevemessick) [reply src/KaggleSwagger.yaml:1541]: Changed to "Number of items per page (default 20)".

File: tests/test_commands.sh
Comment (stevemessick) [reply tests/test_commands.sh:11]: It seems `kaggle/meta-kaggle` is not available when running on localhost, and `lastplacelarry` doesn't have any kernels on localhost. But I found some alternates.

File: docs/KaggleApi.md
Comment (stevemessick) [reply docs/KaggleApi.md:45]: I think this is incorrect and I'm not sure why it works. The CLI does not permit framework and instance to be omitted. I'm changing it to allow the version number to be optional, though. (MT PR linked is below.)

File: kaggle/__init__.py
Comment (stevemessick) [reply kaggle/__init__.py:1]: It was removed as part of file generation, not intentionally.

File: docs/KaggleApi.md
Comment (stevemessick) [reply docs/KaggleApi.md:40]: https://github.com/Kaggle/kaggleazure/pull/29404 adds support for using the latest version if none is specified.

File: kaggle/__init__.py
Comment (stevemessick) [reply kaggle/__init__.py]: The installation of `/tmp/autogen.sh` failed when running in the container. I added instructions to the readme to work around that problem.

**Review by jplotts — APPROVED** [review]

**Pull request merged**
